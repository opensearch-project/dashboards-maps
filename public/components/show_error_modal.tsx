/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import {
  EuiSmallButton,
  EuiButton,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiCodeBlock,
} from '@elastic/eui';

export interface ShowErrorModalProps {
  modalTitle: string;
  modalBody: string;
  buttonText: string;
}

const ShowErrorModal = (props: ShowErrorModalProps) => {
  const [isModalVisible, setIsModalVisible] = useState(false);

  const closeModal = () => setIsModalVisible(false);
  const showModal = () => setIsModalVisible(true);

  let modal;

  if (isModalVisible) {
    modal = (
      <EuiModal onClose={closeModal}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>
            <h1>{props.modalTitle}</h1>
          </EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiCodeBlock isCopyable>{props.modalBody}</EuiCodeBlock>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiSmallButton
            id="closeModal"
            aria-label="closeModal"
            data-testid="closeModal"
            onClick={closeModal}
            fill
          >
            Close
          </EuiSmallButton>
        </EuiModalFooter>
      </EuiModal>
    );
  }

  return (
    <div id="showModalOption" aria-label="showModalOption" data-testid="showModalOption">
      <EuiButton onClick={showModal} size="s">
        {props.buttonText}
      </EuiButton>
      {modal}
    </div>
  );
};

export { ShowErrorModal };
