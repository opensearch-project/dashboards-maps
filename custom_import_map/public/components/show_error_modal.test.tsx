/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import * as React from 'react';
import { ShowErrorModal } from './show_error_modal';
import renderer, { act } from 'react-test-renderer';
import { fireEvent } from '@testing-library/dom';
import { screen, render } from '@testing-library/react';

describe('display error modal', () => {
  const props = {
    modalTitle: 'testModalTitle',
    modalBody: 'testModalBody',
    buttonText: 'testButtonText',
  };

  it('renders the error modal based on the props passed', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<ShowErrorModal {...props} />);
    });
    expect(tree.toJSON().props.id).toBe('showModalOption');
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders the error modal when showModal button is clicked', () => {
    render(<ShowErrorModal {...props} />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    const closeButton = screen.getByTestId('closeModal');
    fireEvent.click(closeButton);
  });
});
