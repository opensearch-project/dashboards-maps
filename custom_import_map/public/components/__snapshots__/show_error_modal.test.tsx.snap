// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display error modal renders the error modal based on the props passed 1`] = `
<div
  aria-label="showModalOption"
  data-testid="showModalOption"
  id="showModalOption"
>
  <button
    className="euiButton euiButton--primary euiButton--small"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "minWidth": undefined,
      }
    }
    type="button"
  >
    <span
      className="euiButtonContent euiButton__content"
    >
      <span
        className="euiButton__text"
      >
        testButtonText
      </span>
    </span>
  </button>
</div>
`;
